### ISSUE TITLE
Could not post to MS Teams sometimes

### ISSUE THREAD

**Issue by woodylic** [justinbarrick/fluxcloud#40]
I installed fluxcloud as sidecar and only enable msteam exporter.

For the first time the pod is created, it always failed to send message, the error message is:
"Could not post to MS Teams: context canceled."

However, when I delete the pod and re-created by replicaset, the problem is gone and the message can be sent successfully.

**Comment by justinbarrick** [comment]
@demikl any ideas on this one?

**Comment by demikl** [comment]
No idea, I'm not experiencing this bug.

**Comment by justinbarrick** [comment]
@woodylic please make sure that you have properly configured your teams endpoints and that the pods in your cluster are able to reach it.

**Comment by woodylic** [comment]
@justinbarrick I will try to re-produce tomorrow. I assume the configuration is correct because I made it work by just destroying the pod and wait the rs to re-create another one, I didn't make any other change. Anyway, will post the update if I found anything.

**Comment by justinbarrick** [comment]
I suspect if recreating the pod fixed it, it was a networking issue with the pod?

**Comment by justinbarrick** [comment]
@woodylic would you try this image and see if it improves things? `justinbarrick/fluxcloud:set-timeouts-19048d3a`

**Comment by woodylic** [comment]
@justinbarrick sorry that didn't have time to try the image, but you are correct that it's a timeout issue. By analyzing the flux log, I found that 2 of 3 events were not sent because of timeout. I also found that sometimes the loop component failed to poll git because hostname cannot be resolved. So your assumption should be correct that it is a network issue. I'm going to check with the cloud provider. Thanks.
